src/lib/consentHandler/consentHandler.js

```diff
@@ -0,0 +1,154 @@
+        let proxy;
+        try{
+            fCmp = window[driver.cmpApi] || window.top[driver.cmpApi];
+        }
+        catch (e) {
+            // noop
+        }
+
+        if (typeof fCmp === 'function') {
+            proxy = fCmp;
+        }
+        else if (window.$sf && window.$sf.ext && window.$sf.ext[driver.safeframeCall]) {
+            proxy = this.callSafeFrame;
+        }
+        else {
+            const frame = findCmpFrame(driver.locatorFrame);
+            if (frame) {
+                const tmp = new FrameProxy(frame, driver);
+                proxy = function (cmd,args,callback) {tmp.call(cmd,args,callback);};
+            }
+        }
+        return proxy;
+    }
+
+    consentEnabled(){
+        return(!!this.cmpProxy);
+    }
+
+    /**
+     * Calls cmp thru safe frame.
+     * @param {string} commandName Name of the cmp request
+     * @param arg
+     * @param {function} callback Callback function
+     */
+    callSafeFrame(commandName, arg, callback) {
+        function sf_callback(msgName, data) {
+
+            if(msgName === this.driver.returnMsgName){
+                let response = data;
+                if(this.driver.getSafeFrameData){
+                    response = this.driver.getSafeFrameData(data);
+                }
+                callback(response, data.success);
+            }
+        }
+
+        window.$sf.ext.register(300, 250, sf_callback);
+        window.$sf.ext[this.driver.safeframeCall](commandName);
+    }
+
+    getCmpConsent(callback, timeout){
+        sendCmpRequests(this.cmpProxy, this.driver.getConsentCmd(), (result, success)=>{
+            if(typeof callback == 'function'){
+                if(success){
+                    callback(this.driver.getConsentData(result), success);
+                }
+                else{
+                    callback(result, success);
+                }
+            }
+        }, timeout);
+    }
+
+    getStaticConsent(callback){
+        callback(this.driver.getConsentData(),true);
+    }
+
+    /**
+     * Entry point to determine driver.
+     * @param {function} callback Function to call after driver has been determined
+     */
+    checkConsent(callback) {
+        if(this.cmpProxy) {
+            this.getConsent((result, success) => {
+                if (success) {
+                    callback(result, true);
+                } else if (this.config.alwaysCallback) {
+                    callback(result, false);
+                }
+
+            }, this.config.timeout);
+        }
+        else{
+            const empty = {gdpr: null, gdpr_consent: null, hasSiteConsent: true};
+            callback(empty, true);
+        }
+    }
+
+    /**
+     * Determine if we have consent
+     */
+    hasSiteConsent(callback){
+        this.checkConsent( (consent, success) => {
+            let hasConsent = false;
+            if(success){
+                hasConsent = consent.hasSiteConsent;
+            }
+            callback(hasConsent);
```

Comment:
I think the callback should also take _success_ as the second flag.

Reply:
ok